src/Microsoft.AspNetCore.Routing.Abstractions/DispatcherDataSource.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNetCore.Routing
+{
+    public abstract class DispatcherDataSource : IEndpointCollectionProvider
```

Comment:
I left out `Address` for now. I need to learn more about the concept of addresses.